libs/dh/shared/feature-authorization/src/lib/scope.service.ts

```diff
@@ -16,18 +16,114 @@
  */
 
 import { Inject, Injectable } from '@angular/core';
+import { MsalBroadcastService, MsalService } from '@azure/msal-angular';
+import {
+  AuthenticationResult,
+  EventMessage,
+  EventType,
+} from '@azure/msal-browser';
 import {
   DhB2CEnvironment,
   dhB2CEnvironmentToken,
 } from '@energinet-datahub/dh/shared/environments';
+import { DhFeatureFlagsService } from '@energinet-datahub/dh/shared/feature-flags';
+import { filter } from 'rxjs';
+import { ScopeStorage, scopeStorageToken } from './scope-storage';
+
+export const actorScopesKey = 'actor-scopes';
+export const activeActorScopeKey = 'active-actor-scope';
+export const actorScopesClaimsKey = 'extn.actors';
 
 @Injectable({ providedIn: 'root' })
 export class ScopeService {
+  localAccountId = '';
+
   constructor(
-    @Inject(dhB2CEnvironmentToken) private config: DhB2CEnvironment
-  ) {}
+    @Inject(dhB2CEnvironmentToken) private config: DhB2CEnvironment,
+    private msalBroadcastService: MsalBroadcastService,
+    private authService: MsalService,
+    private featureFlagService: DhFeatureFlagsService,
+    @Inject(scopeStorageToken) private scopeStorage: ScopeStorage
+  ) {
+    this.msalBroadcastService.msalSubject$
+      .pipe(
+        filter(
+          (msg: EventMessage) =>
+            msg.eventType === EventType.ACQUIRE_TOKEN_SUCCESS
+        )
+      )
+      .subscribe((e: EventMessage) => {
+        const payload = e.payload as AuthenticationResult;
+        const account = payload.account;
+
+        if (!account) {
+          this.clearScopeRelatedItems();
+          return;
+        }
+
+        this.localAccountId = account.localAccountId;
+        this.authService.instance.setActiveAccount(account);
+
+        const userId = account.localAccountId;
+        const actors =
+          account.idTokenClaims &&
+          (account.idTokenClaims[actorScopesClaimsKey] as string[]);
+        const scopes = this.getActorClaims(actors);
+
+        this.scopeStorage.setItem(userId + actorScopesKey, scopes.join(' '));
```

Comment:
suggestion: using the `userId + actorScopesKey` approach results in `353d36a1-c62f-4905-8318-842b009cd008actor-scopes` key. Since the `userId` also contains letters, the `actorScopesKey` value (or a part of it) can be confused as being part of the `userId` string. Consider separating the two. It could be by dot `.`. Can see that MSAL is using this approach (`msal.88e5d356-0c71-49e9-b260-d0629f3c0445.active-account`).

This comment is also related to all other places where `userId + actorScopesKey | actorScopesKey` is used. To make this easier, the logic can be moved to a method, so it can also be used in tests.

Reply:
Fixed